### PR TITLE
Added interbench test  to avocado

### PR DESCRIPTION
**Pull request by PraveenPenguin**
Signed-off-by: praveen@linux.vnet.ibm.com
added interbench Test case

### CODE REVIEW

File: generic/interbench.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Praveen K Pandey <praveen@linux.vnet.ibm.com>
+#
+# Based on code by Brandon Philips <brandon@ifup.org>
+#   copyright: 2008 ifup.org
```
Comment (ldoktor) [inline generic/interbench.py:18]: Well I can see his first commit to be in `2006` fc9892da4993edc4b7badfa93afe18fac578a1f4 and the `ifup.org` is Brandon's custom page. From https://www.linkedin.com/in/brandonphilips I believe the commit was made while he worked for `IBM`, but I can't guarantee that.

File: generic/interbench.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        Build interbench
+        Source:
+        http://www.tux.org/pub/kernel/people/ck/apps/interbench/interbench-0.30.tar.bz2
+        '''
+        sm = SoftwareManager()
+        if not sm.check_installed("gcc") and not sm.install("gcc"):
+            self.error("Gcc is needed for the test to be run")
+        tarball = self.fetch_asset('http://www.tux.org'
+                                   '/pub/kernel/people/ck/apps/interbench/'
+                                   'interbench-0.30.tar.bz2')
```
Comment (ldoktor) [inline generic/interbench.py:51]: nitpick: The current version is 0.31: http://ck.kolivas.org/apps/interbench/interbench-0.31.tar.bz2

File: generic/interbench.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        data_dir = os.path.abspath(self.datadir)
+        archive.extract(tarball, self.srcdir)
+        version = os.path.basename(tarball.split('.tar.')[0])
+        self.srcdir = os.path.join(self.srcdir, version)
+
+        # Patch for make file
+        os.chdir(self.srcdir)
+        p1 = 'patch -p1 < %s/%s' % (data_dir, 'makefile_fix.patch')
```
Comment (ldoktor) [inline generic/interbench.py:59]: please use `os.path.join`

File: generic/interbench.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+        process.run(p1, shell=True)
+
+        build.make(self.srcdir)
+
+    def test(self):
+        args = self.params.get('arg', default='')
+
+        args += ' c'
+        process.system("%s/interbench -m 'run ' %s" % (self.srcdir, args))
```
Comment (ldoktor) [inline generic/interbench.py:69]: well the `-m` option only adds `Comment: ...` to the log. Is it necessary? Running it without any params works also fine.

Comment (ldoktor) [inline generic/interbench.py:69]: Although it gives warning when not used as root and with `-r` it requires root. I'd suggest using `sudo` here, but take it only for consideration, I'm not 100% sure that's what users expect.

File: generic/interbench.py.data/makefile_fix.patch [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+-CFLAGS=-W -Wall -g -O2 -s -pipe
+-LDFLAGS=-lrt -lm
++CC ?= gcc
++CFLAGS=-W -Wall -g -lpthread -O2 -s -pipe
++LDFLAGS=-lrt -lm -lpthread
+ 
+-interbench: interbench.o hackbench.o
+-interbench.o: interbench.c
+-hackbench.o: hackbench.c
++default :
```
Comment (ldoktor) [inline generic/interbench.py.data/makefile_fix.patch:14]: there are some empty spaces, but sure, it works... Btw I sent the patch to Con so hopefully he'll include it in the next release (no response yet)

**Comment by ldoktor** [comment]
Hello @PraveenPenguin, this looks quite good. I found a little license problem and a few nitpicks.

**Comment by PraveenPenguin** [comment]
@ldoktor Thanks for looking , 
made change and pushed  #83 
Thanks

**Pull request closed without merging**